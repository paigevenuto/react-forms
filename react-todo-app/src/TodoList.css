.TodoList {
    width: 50%;
    display: contents;
    margin: auto;
}

.TodoList li button {
    margin-left: 20px;
    background-color: red;
    color: white;
    font-weight: bold;
    border-radius: 1em;
}
