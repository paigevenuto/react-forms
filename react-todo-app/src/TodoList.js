import React, { useState } from "react";
import NewTodoForm from "./NewTodoForm";
import "./TodoList.css";
import { v4 as uuid } from "uuid";
import _ from "lodash";

function TodoList() {
  const [todos, setTodos] = useState([]);

  const addTodo = (formData) => {
    const { task } = formData;
    const id = uuid();
    const todosClone = _.cloneDeep(todos);
    todosClone.push({
      task,
      id,
    });
    setTodos(todosClone);
  };

  const delTodo = (evt) => {
    const todosClone = todos.filter(
      (todo) => todo.id !== evt.target.parentElement.id
    );
    setTodos(todosClone);
  };

  return (
    <ul className="TodoList">
      <NewTodoForm addTodo={addTodo} />
      {todos.map((todo) => (
        <li id={todo.id}>
          {todo.task}
          <button onClick={delTodo}>X</button>
        </li>
      ))}
    </ul>
  );
}

export default TodoList;
