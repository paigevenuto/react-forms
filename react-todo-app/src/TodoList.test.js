import React from "react";
import { render, fireEvent, getNodeText } from "@testing-library/react";
import TodoList from "./TodoList";

// smoke test
it("renders without crashing", function () {
  render(<TodoList />);
});

// snapshot test
it("matches snapshot", function () {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

it("creates a new todo", function () {
  const { queryByText, queryByTestId } = render(<TodoList />);

  const taskInput = queryByTestId("test-input");
  const submitBtn = queryByText("Add Task");
  fireEvent.change(taskInput, { target: { value: "feed my cat" } });

  expect(taskInput.value).toBe("feed my cat");
  fireEvent.click(submitBtn);
  expect(taskInput.value).toBe("");

  let todoItem = queryByTestId("todo");
  todoItem = getNodeText(todoItem);
  expect(todoItem).toContain(`feed my cat`);
});
