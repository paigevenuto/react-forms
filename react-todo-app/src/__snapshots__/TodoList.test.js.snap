// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <ul
    class="TodoList"
    id="TodoList"
  >
    <form
      class="TodoForm"
    >
      <input
        data-testid="test-input"
        name="task"
        required=""
        value=""
      />
      <button>
        Add Task
      </button>
    </form>
  </ul>
</DocumentFragment>
`;
